app/src/androidTest/java/ch/epfl/balelecbud/util/database/MockDatabaseWrapper.java

```diff
@@ -162,16 +166,64 @@ private Field getField(MyQuery.WhereClause clause, Class clazz) {
         return field;
     }
 
-    private boolean evaluateResult(MyQuery.WhereClause.Operator op, int resultOfComparison){
-        switch (op){
-            case LESS_THAN: return resultOfComparison < 0;
-            case LESS_EQUAL: return resultOfComparison <= 0;
-            case GREATER_THAN: return resultOfComparison > 0;
-            case GREATER_EQUAL: return resultOfComparison >= 0;
-            default : return resultOfComparison == 0;
+    private boolean evaluateResult(MyQuery.WhereClause.Operator op, int resultOfComparison) {
+        switch (op) {
+            case LESS_THAN:
+                return resultOfComparison < 0;
+            case LESS_EQUAL:
+                return resultOfComparison <= 0;
+            case GREATER_THAN:
+                return resultOfComparison > 0;
+            case GREATER_EQUAL:
+                return resultOfComparison >= 0;
+            default:
+                return resultOfComparison == 0;
         }
     }
 
+    @Override
+    public CompletableFuture<List<String>> queryIds(MyQuery query) {
+        Map<String, Map<String, Boolean>> mapToQuery = getMapToQuery(query.getCollectionName());
+        List<Map<String, Boolean>> filteredValues = filterMapValues(mapToQuery, query.getWhereClauses());
+        Set<String> filteredKeys = filterKeys(mapToQuery, filteredValues);
+        return CompletableFuture.completedFuture(new LinkedList<>(filteredKeys));
+    }
+
+    private Map<String, Map<String, Boolean>> getMapToQuery(String name) {
+        switch (name) {
+            case DatabaseWrapper.FRIEND_REQUESTS_PATH:
+                return friendRequests;
+            case DatabaseWrapper.FRIENDSHIPS_PATH:
+                return friendships;
+            default:
+                throw new IllegalArgumentException("Unsupported collection name " + name);
+        }
+    }
+
+    private List<Map<String, Boolean>> filterMapValues(Map<String, Map<String, Boolean>> mapToFilter, List<MyQuery.WhereClause> clauses) {
+        List<Map<String, Boolean>> values = new LinkedList<>(mapToFilter.values());
+        for (MyQuery.WhereClause clause : clauses) {
+            List<Map<String, Boolean>> newValues = new LinkedList<>();
```

Comment:
You could use
`values = values.stream().filter(oldValue -> oldValue.getOrDefault(clause.getLeftOperand(), false)).collect(Collectors.toList());`

Reply:
I personally agree and like this functional style but not sure about the rest of the team